benches/plonk.rs

```diff
@@ -247,61 +244,74 @@ fn bench_with_k(name: &str, k: u32, c: &mut Criterion) {
+        };
 
-    let prover_name = name.to_string() + "-prover";
-    let verifier_name = name.to_string() + "-verifier";
+        let mut transcript = Blake2bWrite::<_, _, Challenge255<_>>::init(vec![]);
+        create_proof(params, pk, &[circuit], &[&[]], &mut transcript)
+            .expect("proof generation should not fail");
+        transcript.finalize()
+    }
 
-    c.bench_function(&prover_name, |b| {
-        b.iter(|| {
-            let circuit: MyCircuit<Fp> = MyCircuit {
-                a: Some(Fp::rand()),
-                k,
-            };
+    fn verifier(params: &Params<EqAffine>, vk: &VerifyingKey<EqAffine>, proof: &[u8]) {
+        let msm = params.empty_msm();
+        let mut transcript = Blake2bRead::<_, _, Challenge255<_>>::init(proof);
+        let guard = verify_proof(params, vk, msm, &[&[]], &mut transcript).unwrap();
+        let msm = guard.clone().use_challenges();
+        assert!(msm.eval());
+    }
 
-            // Create a proof
-            let mut transcript = Blake2bWrite::<_, _, Challenge255<_>>::init(vec![]);
-            create_proof(&params, &pk, &[circuit], &[&[]], &mut transcript)
-                .expect("proof generation should not fail")
-        });
-    });
-
-    let circuit: MyCircuit<Fp> = MyCircuit {
-        a: Some(Fp::rand()),
-        k,
-    };
-
-    // Create a proof
-    let mut transcript = Blake2bWrite::<_, _, Challenge255<_>>::init(vec![]);
-    create_proof(&params, &pk, &[circuit], &[&[]], &mut transcript)
-        .expect("proof generation should not fail");
-    let proof = transcript.finalize();
-
-    c.bench_function(&verifier_name, |b| {
-        b.iter(|| {
-            let msm = params.empty_msm();
-            let mut transcript = Blake2bRead::<_, _, Challenge255<_>>::init(&proof[..]);
-            let guard = verify_proof(&params, pk.get_vk(), msm, &[&[]], &mut transcript).unwrap();
-            let msm = guard.clone().use_challenges();
-            assert!(msm.eval());
-        });
-    });
-}
+    let k_range = 8..=16;
 
-fn criterion_benchmark(c: &mut Criterion) {
-    bench_with_k("plonk-k=8", 8, c);
-    bench_with_k("plonk-k=9", 9, c);
-    bench_with_k("plonk-k=10", 10, c);
-    bench_with_k("plonk-k=11", 11, c);
-    bench_with_k("plonk-k=12", 12, c);
-    bench_with_k("plonk-k=13", 13, c);
-    bench_with_k("plonk-k=14", 14, c);
-    bench_with_k("plonk-k=15", 15, c);
-    bench_with_k("plonk-k=16", 16, c);
+    let mut keygen_group = c.benchmark_group("plonk-keygen");
+    keygen_group.sample_size(10);
+    for k in k_range.clone() {
+        keygen_group.bench_with_input(BenchmarkId::from_parameter(k), &k, |b, &k| {
+            b.iter(|| keygen(k));
+        });
+    }
+    keygen_group.finish();
+
+    let mut prover_group = c.benchmark_group("plonk-prover");
+    prover_group.sample_size(10);
+    for k in k_range.clone() {
+        let (params, pk) = keygen(k);
+
+        prover_group.bench_with_input(
+            BenchmarkId::from_parameter(k),
```

Comment:
@ebfull this change will create benchmark names like `plonk-prover/11`, which assumes this benchmark will only vary `k`. Does that seem likely to change at all, or should I alter the naming to make it `plonk-prover/k/11`?

Reply:
This seems fine to me. It doesn't preclude adding other parameters later.

Reply:
Yeah, we can always go into the `gh-pages` branch and rename the existing benchmarks to preserve past data.